plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableProperties.java

```diff
@@ -252,6 +297,48 @@ private static void validateFormatVersion(int version)
         }
     }
 
+    public static void validateCompression(IcebergFileFormat fileFormat, Optional<HiveCompressionCodec> compressionCodec, OptionalInt compressionLevel)
+    {
+        if (compressionCodec.isPresent()) {
+            if (!isCompressionCodecSupportedForFormat(fileFormat, compressionCodec.get())) {
+                throw new TrinoException(NOT_SUPPORTED, format("Compression codec LZ4 not supported for %s", fileFormat.humanName()));
+            }
+        }
+        if (compressionLevel.isPresent()) {
+            if (compressionCodec.isEmpty()) {
+                throw new TrinoException(INVALID_TABLE_PROPERTY, "write_compression must be set when compression_level is set");
+            }
+            else {
+                if (!(VALID_ICEBERG_FILE_FORMATS_FOR_COMPRESSION_LEVEL_PROPERTY.contains(fileFormat)
```

Comment:
I wonder whether this branch would actually be easier to read and understand as an enum switch statement. That way we would fail to compile if a new enum were added and we would know we needed to add new handling at that point.

Reply:
done